app/main.py

```diff
@@ -1,34 +1,63 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self,
+                 row: int,
+                 column: int,
+                 is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = []
+        if self.start[0] == self.end[0] and self.start[1] == self.end[1]:
+            self.decks = [Deck(self.start[0], self.start[1])]
+            return
+        if self.start[0] == self.end[0]:
+            for y in range(self.start[1], self.end[1] + 1):
+                self.decks.append(Deck(self.start[0], y))
+        if self.start[1] == self.end[1]:
+            for x in range(self.start[0], self.end[0] + 1):
+                self.decks.append(Deck(x, self.start[1]))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self,
+                 row: int,
+                 column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self,
+             row: int,
+             column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+        for deck in self.decks:
+            if deck.is_alive:
+                break
+        else:
```

Comment:
Using `else` with `for` and `while` loops is neither very intuitive* nor commonly used. I would advise against using it. However, I may be wrong.

\* For an average Python developer without any background in low-level languages